### PR TITLE
dev-python/shiboken,dev-python/pyside,dev-python/pyside-tools: new packages

### PR DESCRIPTION
**Pull request by Perlovka**
These packages are dependencies for FreeCAD.

### CODE REVIEW

**Comment by Perlovka** [comment]
@Pesa , @kensington, @Chiitoo  please review.

**Comment by Pesa** [comment]
are these derived from the ebuilds in https://github.com/gentoo/qt ?

**Comment by Perlovka** [comment]
@Pesa, yes.

**Comment by mgorny** [comment]
@gentoo/qt

File: dev-python/pyside/pyside-5.11.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+# See "sources/pyside2/PySide2/licensecomment.txt" for licensing details.
+LICENSE="|| ( GPL-2 GPL-3+ LGPL-3 )"
+SLOT="2"
+KEYWORDS="~amd64 ~x86"
+
+IUSE="3d charts concurrent datavis declarative designer gui help location multimedia
+	network opengl positioning printsupport script scripttools scxml sensors speech
+	sql svg test testlib webchannel webengine websockets widgets x11extras xmlpatterns"
+
```
Comment (waebbl) [inline dev-python/pyside/pyside-5.11.1.ebuild]: Nice to see, you have added USE flags for the missing modules from live ebuild. But don't you need to add them in REQUIRED_USE as well?

File: dev-python/pyside/pyside-5.11.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+REQUIRED_USE="${PYTHON_REQUIRED_USE}
```
Comment (waebbl) [inline dev-python/pyside/pyside-5.11.1.ebuild]: `3d? ( gui network )`

File: dev-python/pyside/pyside-5.11.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	charts? ( widgets )
+	datavis? ( gui )
+	declarative? ( gui network )
+	designer? ( widgets )
+	help? ( widgets )
```
Comment (waebbl) [inline dev-python/pyside/pyside-5.11.1.ebuild]: `location? ( positioning )`

File: dev-python/pyside/pyside-5.11.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	multimedia? ( gui network )
+	opengl? ( widgets )
+	printsupport? ( widgets )
+	scripttools? ( gui script widgets )
```
Comment (waebbl) [inline dev-python/pyside/pyside-5.11.1.ebuild]: speech? ( multimedia )

**Comment by waebbl** [comment]
Wouldn't it be clearer to depend the three packages on the versioned `dev-qt/*` packages, like the other qt packages depend on the same version?
For example use `~dev-qt/qtcore-${PV}:5=` instead of just `dev-qt/qtcore:5`

File: dev-python/pyside/pyside-5.11.1.ebuild (hunk 1 above)
Comment (Pesa) [reply dev-python/pyside/pyside-5.11.1.ebuild]: yes please

**Comment by Pesa** [comment]
> Wouldn't it be clearer to depend the three packages on the versioned dev-qt/* packages, like the other qt packages depend on the same version?

It's not about "clarity", deps need to be *correct*. If there's no need for a `~`-style dependency, don't add it, it unnecessarily complicates upgrades.

File: dev-python/shiboken/shiboken-5.11.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+# version 1.0 of a Qt-specific exception enabling shiboken2 output to be
+# arbitrarily relicensed. (TODO)
+LICENSE="|| ( GPL-2 GPL-3+ LGPL-3 ) GPL-3"
+SLOT="2"
+KEYWORDS="~amd64 ~x86"
+IUSE="numpy test"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+DEPEND="${PYTHON_DEPS}
+	numpy? ( dev-python/numpy[${PYTHON_USEDEP}] )
```
Comment (Pesa) [inline dev-python/shiboken/shiboken-5.11.1.ebuild]: we usually sort conditional deps after unconditional ones, IOW move this line last

File: dev-python/shiboken/shiboken-5.11.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,90 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
```
Comment (Pesa) [inline dev-python/shiboken/shiboken-5.11.1.ebuild]: upstream doc says python 3.4 is not supported

File: dev-python/shiboken/shiboken-5.11.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,90 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (Pesa) [inline dev-python/shiboken/shiboken-5.11.1.ebuild]: use EAPI=7 if possible

File: dev-python/shiboken/shiboken-5.11.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+DEPEND="${PYTHON_DEPS}
+	numpy? ( dev-python/numpy[${PYTHON_USEDEP}] )
+	dev-libs/libxml2
+	dev-libs/libxslt
+	dev-qt/qtcore:5
+	dev-qt/qtxml:5
+	dev-qt/qtxmlpatterns:5
+	sys-devel/clang:="
```
Comment (Pesa) [inline dev-python/shiboken/shiboken-5.11.1.ebuild]: clang 5.0 or higher is required (according to upstream docs)

File: dev-python/shiboken/shiboken-5.11.1.ebuild [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/pyside-setup-everywhere-src-${PV}/sources/shiboken2"
```
Comment (Pesa) [inline dev-python/shiboken/shiboken-5.11.1.ebuild]: declare `MY_P=pyside-setup-everywhere-src-${PV}` and use it here and in `SRC_URI`

File: dev-python/shiboken/shiboken-5.11.1.ebuild [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+src_prepare() {
+	#FIXME: File an upstream issue requesting a sane way to disable NumPy support.
+	if ! use numpy; then
+		sed -i -e '/print(os\.path\.realpath(numpy))/d' libshiboken/CMakeLists.txt || die
+	fi
+
+	if use prefix; then
+		cp "${FILESDIR}"/rpath.cmake . || die
```
Comment (Pesa) [inline dev-python/shiboken/shiboken-5.11.1.ebuild]: is this "patch" still needed? do we even care about prefix anymore?

File: dev-python/shiboken/shiboken-5.11.1.ebuild [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+		sed -i -e '1iinclude(rpath.cmake)' CMakeLists.txt || die
+	fi
+
+	eapply "${FILESDIR}"/${PN}-5.11.1-fix-warnings.patch
```
Comment (Pesa) [inline dev-python/shiboken/shiboken-5.11.1.ebuild]: why isn't this in `PATCHES`?

File: dev-python/shiboken/shiboken-5.11.1.ebuild [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+src_test() {
+	python_foreach_impl cmake-utils_src_test
+}
+
+src_install() {
+	installation() {
+		cmake-utils_src_install
+		mv "${ED}"usr/$(get_libdir)/pkgconfig/${PN}2{,-${EPYTHON}}.pc || die
```
Comment (Pesa) [inline dev-python/shiboken/shiboken-5.11.1.ebuild]: meh... upstream still doesn't support parallel installations for multiple python versions?

File: dev-python/shiboken/shiboken-5.11.1.ebuild (hunk 7 above)
Comment (waebbl) [reply dev-python/shiboken/shiboken-5.11.1.ebuild]: this is not possible yet, due to cmake-utils not supporting `EAPI=7`

**Comment by waebbl** [comment]
> > Wouldn't it be clearer to depend the three packages on the versioned dev-qt/* packages, like the other qt packages depend on the same version?
> 
> It's not about "clarity", deps need to be _correct_. If there's no need for a `~`-style dependency, don't add it, it unnecessarily complicates upgrades.

I agree, but wouldn't it confuse users, if all their qt related packages need to be of the same version, but these do not? Technically they don't seem to need their version counterparts on qt packages, they need >=qt-5.7, so a 'correct' dep would be `>=dev-qt/qt-*-5.7:5`. That's why I said for 'clarity'. From what I see, upstream puts them into their usual versioning scheme, with 5.11.2 being the current release, as it is for other Qt packages. So I would vote for bringing them on par with the other Gentoo Qt packages.

**Comment by Perlovka** [comment]
> Technically they don't seem to need their version counterparts on qt packages, they need >=qt-5.7, so a 'correct' dep would be `>=dev-qt/qt-*-5.7:5`.

Minimal Qt version in tree is 5.9, do we still need such dependency?

**Comment by waebbl** [comment]
> > Technically they don't seem to need their version counterparts on qt packages, they need >=qt-5.7, so a 'correct' dep would be `>=dev-qt/qt-*-5.7:5`.
> 
> Minimal Qt version in tree is 5.9, do we still need such dependency?

This was only a theoretical construct, based on the content of the CMakeLists.txt file. IMO it wouldn't make sense to depend on 5.7, but better wait for someone to review it.

**Comment by Pesa** [comment]
> I agree, but wouldn't it confuse users, if all their qt related packages need to be of the same version, but these do not?

No.

> Technically they don't seem to need their version counterparts on qt packages, they need >=qt-5.7, so a 'correct' dep would be >=dev-qt/qt-*-5.7:5

I'd be fine with a `>=5.9` or even `>=5.11` dependency, even if it's higher than what's actually required. But I'm strongly against a `~` dep when there's no need for it.

> doesn't being essential imply they shouldn't be deselectable? 

No, it doesn't. The upstream distinction between "essential" and "add-on" is almost completely meaningless for us.

**Comment by Perlovka** [comment]
Fixed, bumped and rebased.

File: dev-python/shiboken/shiboken-5.11.1.ebuild (hunk 12 above)
Comment (Perlovka) [reply dev-python/shiboken/shiboken-5.11.1.ebuild]: Sadly, I don't know how to achieve parallel installation here, so left this as is.

**Comment by Pesa** [comment]
Would you mind sending a PR to https://github.com/gentoo/qt to get a little more testing before we move these to the main tree?

**Comment by Pesa** [comment]
This PR is obsolete. A new attempt is being made in https://github.com/gentoo/qt/pull/194 and subsequent PRs.

**Pull request closed without merging**